tuplex/adapters/cpython/src/PythonHelpers.cc

```diff
@@ -682,19 +689,27 @@ namespace python {
 
             std::vector<tuplex::Field> v;
             for(unsigned i = 0; i < numElements; ++i) {
-                v.push_back(pythonToField(PyTuple_GetItem(obj, i), type.parameters()[i]));
+                v.push_back(pythonToField(PyTuple_GetItem(obj, i), type.parameters()[i], autoUpcast));
             }
             return tuplex::Field(tuplex::Tuple::from_vector(v));
+        } else if(type.isListType() && type != python::Type::EMPTYLIST) {
+            auto numElements = PyList_Size(obj);
+            auto elementType = type.elementType();
+            std::vector<tuplex::Field> v;
+            for(unsigned i = 0; i < numElements; ++i) {
+                v.push_back(pythonToField(PyList_GetItem(obj, i), elementType, autoUpcast));
```

Comment:
potentially we should PyINC the GetItem item obj, because when we destroy the list ref counts are off then.